specs/ContextMenuRequested.md

```diff
@@ -0,0 +1,383 @@
+
+        /// The coordinates where the right click occured
+        [propget] HRESULT Location([out, retval] POINT* coordinates);
+
+        /// The context that the user selected
+        [propget] HRESULT Context([out, retval] COREWEBVIEW2_SELECTION_CONTEXT* context);
+
+        /// Returns the selected Context Menu Item
+        [propget] HRESULT SelectedItem([out, retval] COREWEBVIEW2_CONTEXT_MENU_ITEM_ID* selectedItem);
+
+        /// Sets the Selected Menu Item to respond to the server
+        [propput] HRESULT SelectedItem([in] COREWEBVIEW2_CONTEXT_MENU_ITEM_ID selectedItem);
+
+        /// Whether the App will draw context menu. False by default, meaning WebView should display default context menu
+        /// If set to true, app developer will handle displaying the Context Menu using the data provided.
+        [propput] HRESULT Handled([out, retval] BOOL* handled);
+
+        /// Sets the Handled property
+        [propget] HRESULT Handled([in] BOOL handled);
+
+        /// Returns an `ICoreWebView2Deferral` object. Use this operation to
+        /// complete the event at a later time.
+        HRESULT GetDeferral([out, retval] ICoreWebView2Deferral ** deferral);
+    }
+```
+
+```c#
+namespace Microsoft.Web.WebView2.Core
+{
+
+    runtimeclass CoreWebView2ContextMenuRequestedEventArgs;
+    runtimeclass CoreWebView2ContextMenuItem;
+    runtimeclass CoreWebView2ContextMenuItemCollection;
+    runtimeclass CoreWebView2ContextMenuItemCollectionIterator;
+
+    enum CoreWebView2ContextMenuItemID {
+        /// Custome type, will include Items brought in by browser extensions
+        Custom = 0,
+
+        /// No Command
+        None = 1,
+
+        /// Page Commands
+        Back = 2,
+        Forward = 3,
+        Reload = 4,
+        Stop = 5,
+        NewWindow = 6,
+        Print = 7,
+
+        /// Clipboard Commands
+        Cut = 8,
+        Copy = 9,
+        Paste = 10,
+        PasteAsPlainText = 11,
+        SelectAll = 12,
+
+        /// Saving
+        SaveLinkAs = 13,
+        SaveImageAs = 14,
+        
+        Inspect = 15,
+        ... /// list is not complete
+    };
+
+    enum CoreWebView2SelectionContext
+    {
+        Image = 0,
+        Link = 1,
+        Textbox = 2,
+        Text = 3,
+        Page = 4,
+        Video = 5,
+        PDF = 6,
+        ... /// list is not complete
+    };
+
+    runtimeclass CoreWebView2ContextMenuRequestedEventArgs
+    {
+        Boolean Handled { get; set; }
+        CoreWebView2SelectionContext Context { get; };
+        CoreWebView2ContextMenuItemCollection MenuItems { get; };
+        CoreWebView2ContextMenuItem SelectedItem { get; set; }
+        Point Location { get; };
+        Windows.Foundation.Deferral GetDeferral();
+    };
+    
+    runtimeclass CoreWebView2ContextMenuItem
+    {
+        String Name { get; }
+        String Shortcut { get; }
+        CoreWebView2ContextMenuItemID ID { get; }
```

Comment:
By our naming conventions it should be 'Id' not 'ID'. If this is the result of our api generator tool let me know and we can open a bug against it.

```suggestion
        CoreWebView2ContextMenuItemId Id { get; }
```